### PR TITLE
internal/status: Avoid meaningless Condition changes

### PR DESCRIPTION
**Pull request by evankanderson**
No longer automatically perform a Status update on _any_ call to AddCondition.

I recently noticed in my cluster that my HTTPRoute objects had updates to `status.parents.conditions[Valid].LastTransitionTime` once per second. In turn, this triggered my own reconcilers to check out the changes, adding load to the cluster and churn on etcd.

I could see an argument for updating the anonymous function in `RouteConditionsAccessor` in `cache.go` instead of `AddCondition` -- I don't have a strong feeling one way or the other.

### CODE REVIEW

**Review by skriss — APPROVED** [review]
LGTM, thanks @evankanderson!

**Comment by youngnick** [comment]
The change makes sense to me, seems like it should be more efficient. I'll wait for @evankanderson to update before pushing the approve button though.

**Comment by youngnick** [comment]
ping @evankanderson - do you still want to go ahead here?

**Pull request closed without merging**
